core/src/test/scala/integration/kafka/api/ProducerSendWhileDeletionTest.scala

```diff
@@ -0,0 +1,86 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one or more
+  * contributor license agreements.  See the NOTICE file distributed with
+  * this work for additional information regarding copyright ownership.
+  * The ASF licenses this file to You under the Apache License, Version 2.0
+  * (the "License"); you may not use this file except in compliance with
+  * the License.  You may obtain a copy of the License at
+  *
+  *    http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+package kafka.api
+
+import kafka.server.KafkaConfig
+import kafka.utils.TestUtils
+import org.apache.kafka.clients.producer.ProducerRecord
+
+import java.util.Properties
+import org.apache.kafka.common.TopicPartition
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+
+import java.nio.charset.StandardCharsets
+
+
+class ProducerSendWhileDeletionTest extends BaseProducerSendTest {
```

Comment:
Out of curiosity, why does this need a new class (as opposed to adding to `PlaintextProducerTest` for example)?

Reply:
I earlier thought of putting it in `PlaintextProducerTest` but the new case needs 2 as RF for reassignments to take place (to increase leader epoch) and disabling `AutoLeaderRebalance`. Modifying the configs for the whole test class would affect the other tests, so decided to go for a new test class.

Reply:
It might be better to extend `IntegrationTestHarness` instead of `BaseProducerSendTest` in this case. `BaseProducerSendTest` contains unit test that will be ran as well as the moment. I don't think that we have to run them with this configuration, or do we?

Reply:
Agreed that extending `BaseProducerSendTest` would run unnecessary tests. Changed now to extend from `IntegrationTestHarness` and verified that the test fails w/o the fix and passes after the fix.